src/nupic/ntypes/Sdr.hpp

```diff
@@ -575,16 +611,44 @@ class SparseDistributedRepresentation : public Serializable
      * @param fractionNoise The fraction of active bits to swap out.  The
      * original and resulting SDRs have an overlap of (1 - fractionNoise).
      *
-     * @param rng The random number generator to draw from, optional.
+     * Both of the following parameters are optional, it neither is given then
+     * the seed 0 is used.
+     * @param seed The seed for the random number generator.
+     * @param rng The random number generator to draw from.
      */
-    void addNoise(Real fractionNoise) {
-        Random rng;
+    void addNoise(Real fractionNoise, UInt64 seed = 0) {
+        Random rng( seed );
```

Comment:
> takes a reference to the RNG 

Ok, that works ok, Thanks. 

>  a.randomize( .02 );
    b.randomize( .02 );
    ASSERT_TRUE( a != b);

Do we want that? I assume it's reasonable behavior. Then, the seed must be always 0 (random), as 
> a.randomize( .02, 42 /*seed*/ ); 

would fail that test. So we should remove the param (sorry for making you add it in the first place, I didn't see this consequence)